### ISSUE TITLE
Un-fixing with javascript

### ISSUE THREAD

**Issue by MarkSizer** [Shira-3749/jquery-scrollfix#1]
I'm trying to use this plugin in a situation where the scroll fix only applies at certain screen resolutions; namely mobile.  I can get the scroll fix to apply correctly, but not remove / detach / unbind if the screen resizes.  Am I simply missing the correct syntax to do this, or is it not supported functionality?

Ideally I'd like to be able to do something like:

```
$(window).resize(function () {
    if ($(this).width() < 768) {
        $('.pageHeader').scrollFix();
    } else {
        $('.pageHeader').scrollUnFix();
    }
});
```

**Comment by shira-374** [comment]
It is supported. You need to utilise the return value of the `scrollFix()` method, like so:

```
var watcher = $('.pageHeader').scrollFix();

$(window).resize(function () {
    if ($(this).width() < 768) {
        // attach
        if (!watcher.attached) {
            watcher.attach();
        }
    } else {
        // detach and unfix
        if (watcher.attached) {
            watcher.detach();
        }
        if (watcher.fixed) {
            watcher.unfix();
            watcher.fixed = false;
        }
    }
});

// initial update
$(window).resize();
```

I will make this easier in a future release.

**Comment by shira-374** [comment]
As promised, this can be done more elegantly since v2.1.0:

``` javascript
$('.pageHeader')
    .on('fix.shira.scrollfix update.shira.scrollfix', function (e) {
        if ($(window).width() >= 768) {
            if (e.watcher.fixed) {
                e.watcher.unfix();
            } else {
                e.preventDefault();
            }
        }
    })
    .scrollFix()
;
```
